runtime/queries/c-sharp/highlights.scm

```diff
@@ -334,14 +354,26 @@
 (lock_statement (identifier) @variable)
 
 ;; Rest
-(member_access_expression) @variable
+(member_access_expression
+  expression: ((identifier) @type
+    (#match? @type "^[A-Z]")))
```

Comment:
Trying to make this look nice in a file like https://github.com/EventStore/EventStore/blob/7ffc4bc215e47398f2f7126f6add55c7dba23cdd/src/EventStore.Core.Tests/ClientAPI/ExpectedVersion64Bit/append_to_stream_with_event_numbers_greater_than_2_billion.cs, I think:

```scm
(member_access_expression
  expression: (identifier) @type @_expr
  name: (identifier) @type @_name
  (#match? @_expr "^[A-Z]")
  (#match? @_name "^[A-Z]"))
(member_access_expression
  expression: (_) @_expr
  name: (identifier) @type @_name
  (#match? @_expr "^[A-Z]")
  (#match? @_name "^[A-Z]"))
(member_access_expression
  expression: ((identifier) @type
    (#match? @type "^[A-Z]")))
```

Works nicely. We might be able to simplify it a little. I don't know C# well enough to know if that highlighting is correct though 😅. What do you think?

Reply:
That is an interesting idea, but it's not quite accurate. In C# member accesses can be properties, fields, types, or constants, and all of them except fields can be PascalCase, so extra context is needed to know which colour to pick. But tree-sitter-c-sharp doesn't seem to provide that context, so it's difficult to accurately colour anything other than the first variable being accessed (which can only be a variable or a type, camelCase and PascalCase respectively).

As a result your code often ends up incorrectly flagging properties and constants as types. That's why I chose to interpret everything except the first variable as a variable, because it felt like the least confusing solution. But I'm still open to better solutions